sql/core/src/main/scala/org/apache/spark/sql/execution/python/TransformWithStateInPandasStateServer.scala

```diff
@@ -146,6 +146,9 @@ class TransformWithStateInPandasStateServer(
 
     while (listeningSocket.isConnected &&
       statefulProcessorHandle.getHandleState != StatefulProcessorHandleState.CLOSED) {
+      if (Thread.currentThread().isInterrupted) {
+        throw new InterruptedException("Thread was interrupted")
```

Comment:
Since interruption is handled in the try/catch block, I think this if statement is not needed.

If we keep the if statement, `StatefulProcessorHandleState` should be set to `CLOSED` here.

Reply:
Good point, or should it be placed inside the `try` block? Can we confirm that
```scala
val version = inputStream.readInt()
        if (version != -1) {
          assert(version == 0)
          val message = parseProtoMessage()
          handleRequest(message)
          outputStream.flush()
        }
```

 can respond to Interrupted? Could you help submit a followup to help to further address this issue?

Reply:
Yea, I think we can remove this check

Reply:
https://github.com/apache/spark/pull/49356 has been created as follow-up.